ts/features/idpay/barcode/saga/index.ts

```diff
@@ -0,0 +1,83 @@
+import * as E from "fp-ts/lib/Either";
+import { pipe } from "fp-ts/lib/function";
+import { SagaIterator } from "redux-saga";
+import { call, put, takeLatest } from "typed-redux-saga/macro";
+import { ActionType } from "typesafe-actions";
+import { TransactionBarCodeResponse } from "../../../../../definitions/idpay/TransactionBarCodeResponse";
+import {
+  CodeEnum,
+  TransactionErrorDTO
+} from "../../../../../definitions/idpay/TransactionErrorDTO";
+import { SagaCallReturnType } from "../../../../types/utils";
+import { withRefreshApiCall } from "../../../fastLogin/saga/utils";
+import { IDPayClient } from "../../common/api/client";
+import { idpayGenerateBarcode } from "../store";
+
+export function* watchIDPayBarcodeSaga(
+  idPayClient: IDPayClient,
+  bearerToken: string
+): SagaIterator {
+  yield* takeLatest(
+    idpayGenerateBarcode.request,
+    handleGenerateBarcode,
+    idPayClient.createBarCodeTransaction,
+    bearerToken
+  );
+}
+
+const genericError: TransactionErrorDTO = {
+  code: CodeEnum.PAYMENT_GENERIC_ERROR,
+  message: "error"
+};
+
+export function* handleGenerateBarcode(
+  createBarCodeTransaction: IDPayClient["createBarCodeTransaction"],
+  bearerToken: string,
+  action: ActionType<typeof idpayGenerateBarcode.request>
+) {
+  const createBarCodeTransactionRequest = createBarCodeTransaction({
+    bearerAuth: bearerToken,
+    body: {
+      initiativeId: action.payload.initiativeId
+    }
+  });
+
+  try {
+    const createBarCodeTransactionResult = (yield* call(
+      withRefreshApiCall,
+      createBarCodeTransactionRequest,
+      action
+    )) as unknown as SagaCallReturnType<typeof createBarCodeTransaction>;
+
+    yield pipe(
+      createBarCodeTransactionResult,
+      E.fold(
+        _error =>
+          put(
+            idpayGenerateBarcode.failure({
+              initiativeId: action.payload.initiativeId,
+              error: genericError
+            })
+          ),
+        response =>
+          put(
+            response.status === 201
+              ? idpayGenerateBarcode.success(
+                  response.value as TransactionBarCodeResponse
```

Comment:
I think we could omit this
```suggestion
                  response.value
```